File: backend/src/main/java/io/nqa/menetlus/entity/Menetlus.java
package io.nqa.menetlus.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "menetlus")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Menetlus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false, length = 11)
    private long personalCode;
    @Column(nullable = false)
    private String email;
    @Column(nullable = false)
    private boolean emailDelivered;
    @Column(nullable = false)
    private String reason;
}

File: frontend/src/app/model/menetlus.ts
export class Menetlus {
  id: number = 0;
  name: string;
  personalCode: number;
  email: string;
  emailDelivered: boolean = false;
  reason: string;

  constructor(
    name: string,
    personalCode: number,
    email: string,
    reason: string
  ) {
    this.name = name;
    this.personalCode = personalCode;
    this.email = email;
    this.reason = reason;
  }
}
